src/dirtyfields/dirtyfields.py

```diff
@@ -151,7 +151,10 @@ def is_dirty(self, check_relationship=False, check_m2m=None):
         return {} != self.get_dirty_fields(check_relationship=check_relationship,
                                            check_m2m=check_m2m)
 
-    def save_dirty_fields(self):
+    def save_dirty_fields(self, full_save_on_add=False):
+        if full_save_on_add and self._state.adding:
```

Comment:
dirtyfields has the established convention that all fields in an unsaved model are dirty. Can you remove the `full_save_on_add` argument and just always call `save() ` when the model is being added.

Reply:
here I thought about the backward compatibility. But if the lib has the established convention, then we can for sure remove the argument

Reply:
Yes backwards compatibility is important, but the current behavior is that Django raises an exception right? I hope that no one is relying on that behavior.

Avoiding the complexity of a new argument is more important to me than the small chance this will break for someone.